desc/configuration.py

```diff
@@ -682,23 +661,42 @@ def NFP(self):
 
     @NFP.setter
```

Comment:
@ddudt What is the use case for changing NFP? If it's something we want to allow we should add extra stuff to this method that updates bases etc.

Reply:
Add logic to update everything that depends on `NFP` (bases, surface, etc.)

Reply:
Temporarily removing this so we can merge the rest in, I'll open another PR to allow changing NFP and some other minor related stuff